src/main/java/com/microsoft/graph/core/models/TokenValidable.java

```diff
@@ -0,0 +1,182 @@
+
+    /**
+     * Sets collection of validation tokens
+     * @param validationTokens tokens
+     */
+    public void setValidationTokens(@Nullable final List<String> validationTokens);
+
+    /**
+     * Returns validation tokens
+     * @return list of tokens
+     */
+    public @Nullable List<String> getValidationTokens();
+
+    /**
+     * Sets collection of encrypted token bearers
+     * @param value collection of encrypted token bearers
+     */
+    public void setValue(@Nullable final List<T> value);
+
+    /**
+     * Get collection of encrypted token bearers
+     * @return encrypted token bearers
+     */
+    public @Nullable List<T> getValue();
+
+    /**
+     * Validates the tokens
+     * @param <U> DecryptableContent
+     * @param <T> EncryptedContentBearer
+     * @param collection collection of encrypted token bearers
+     * @param tenantIds tenant ids
+     * @param appIds app ids
+     * @param keyDiscoveryUrl the JWKS endpoint to use to retrieve signing keys
+     * @return true if the tokens are valid
+     * @throws IllegalArgumentException if one of the tokens are invalid
+     */
+    public static <U extends DecryptableContent, T extends EncryptedContentBearer<U>>
+        boolean areTokensValid(
+            @Nonnull final TokenValidable<U,T> collection,
+            @Nonnull final List<UUID> tenantIds,
+            @Nonnull final List<UUID> appIds,
+            @Nonnull final String keyDiscoveryUrl) {
+
+        Objects.requireNonNull(collection);
+        Objects.requireNonNull(tenantIds);
+        Objects.requireNonNull(appIds);
+        Objects.requireNonNull(keyDiscoveryUrl);
+
+        if (collection.getValidationTokens() == null
+                || collection.getValidationTokens().isEmpty()
+                || collection.getValue().stream().allMatch(x -> x.getEncryptedContent() == null)) {
+            return true;
+        }
+
+        if (tenantIds.isEmpty() || appIds.isEmpty()) {
+            throw new IllegalArgumentException("tenantIds, appIds and issuer formats must be provided");
+        }
+
+        for (final String token : collection.getValidationTokens()) {
+            if (!isTokenValid(token, tenantIds, appIds, keyDiscoveryUrl)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Validates the tokens
+     * @param <U> DecryptableContent
+     * @param <T> EncryptedContentBearer
+     * @param collection collection of encrypted token bearers
+     * @param tenantIds tenant ids
+     * @param appIds app ids
+     * @return true if the tokens are valid
+     */
+    public static <U extends DecryptableContent, T extends EncryptedContentBearer<U>>
+        boolean areTokensValid(
+            @Nonnull final TokenValidable<U,T> collection,
+            @Nonnull final List<UUID> tenantIds,
+            @Nonnull final List<UUID> appIds) {
+
+        final String keyDiscoveryUrl = "https://login.microsoftonline.com/common/discovery/keys";
```

Comment:
the discovery url should be configurable somehow, national clouds use different keys

Reply:
This is already the case as the `areTokensValid` function provides an overload where the `keyDiscoveryUrl` can be passed in as the last parameter. 
This is here since java doesn't work with optional parameter with defaults the same way as other languages so this overload exists to set it to a reasonable default.